composer/utils/checkpoint.py

```diff
@@ -287,7 +287,9 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
             receiver = dist.get_global_rank() != rank_in_first_replica
 
             # Send list of files to all ranks
-            file_list = [sorted(os.listdir(self.destination_path))]
+            file_list = [
```

Comment:
is metadata sent elsewhere?

Reply:
Metadata is downloaded on all ranks and not broadcasted -- it's needed for other things (rng) before broadcast. I removed the line that filters it as well.